src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -491,6 +474,8 @@ - (instancetype)initForSubclassesWithNodeID:(NSNumber *)nodeID controller:(MTRDe
     if (self = [super init]) {
         _lock = OS_UNFAIR_LOCK_INIT;
         _delegates = [NSMutableSet set];
+        _deviceController = controller;
```

Comment:
This part already added in https://github.com/project-chip/connectedhomeip/pull/35110/files

Reply:
Yes, I purposefully made this one exactly like that one, so it will merge cleanly either direction.